Source/WebCore/html/PermissionsPolicy.cpp

```diff
@@ -39,62 +39,171 @@ namespace WebCore {
 
 using namespace HTMLNames;
 
-static ASCIILiteral policyTypeName(PermissionsPolicy::Type type)
+static ASCIILiteral toFeatureNameForLogging(PermissionsPolicy::Feature feature)
 {
-    switch (type) {
-    case PermissionsPolicy::Type::Camera:
+    switch (feature) {
+    case PermissionsPolicy::Feature::Camera:
         return "Camera"_s;
-    case PermissionsPolicy::Type::Microphone:
+    case PermissionsPolicy::Feature::Microphone:
         return "Microphone"_s;
-    case PermissionsPolicy::Type::SpeakerSelection:
+    case PermissionsPolicy::Feature::SpeakerSelection:
         return "SpeakerSelection"_s;
-    case PermissionsPolicy::Type::DisplayCapture:
+    case PermissionsPolicy::Feature::DisplayCapture:
         return "DisplayCapture"_s;
-    case PermissionsPolicy::Type::Gamepad:
+    case PermissionsPolicy::Feature::Gamepad:
         return "Gamepad"_s;
-    case PermissionsPolicy::Type::Geolocation:
+    case PermissionsPolicy::Feature::Geolocation:
         return "Geolocation"_s;
-    case PermissionsPolicy::Type::Payment:
+    case PermissionsPolicy::Feature::Payment:
         return "Payment"_s;
-    case PermissionsPolicy::Type::ScreenWakeLock:
+    case PermissionsPolicy::Feature::ScreenWakeLock:
         return "ScreenWakeLock"_s;
-    case PermissionsPolicy::Type::SyncXHR:
+    case PermissionsPolicy::Feature::SyncXHR:
         return "SyncXHR"_s;
-    case PermissionsPolicy::Type::Fullscreen:
+    case PermissionsPolicy::Feature::Fullscreen:
         return "Fullscreen"_s;
-    case PermissionsPolicy::Type::WebShare:
+    case PermissionsPolicy::Feature::WebShare:
         return "WebShare"_s;
 #if ENABLE(DEVICE_ORIENTATION)
-    case PermissionsPolicy::Type::Gyroscope:
+    case PermissionsPolicy::Feature::Gyroscope:
         return "Gyroscope"_s;
-    case PermissionsPolicy::Type::Accelerometer:
+    case PermissionsPolicy::Feature::Accelerometer:
         return "Accelerometer"_s;
-    case PermissionsPolicy::Type::Magnetometer:
+    case PermissionsPolicy::Feature::Magnetometer:
         return "Magnetometer"_s;
 #endif
 #if ENABLE(WEB_AUTHN)
-    case PermissionsPolicy::Type::PublickeyCredentialsGetRule:
+    case PermissionsPolicy::Feature::PublickeyCredentialsGetRule:
         return "PublickeyCredentialsGet"_s;
 #endif
 #if ENABLE(WEBXR)
-    case PermissionsPolicy::Type::XRSpatialTracking:
+    case PermissionsPolicy::Feature::XRSpatialTracking:
         return "XRSpatialTracking"_s;
 #endif
-    case PermissionsPolicy::Type::PrivateToken:
+    case PermissionsPolicy::Feature::PrivateToken:
         return "PrivateToken"_s;
+    case PermissionsPolicy::Feature::Invalid:
+        return "Invalid"_s;
     }
     ASSERT_NOT_REACHED();
     return ""_s;
 }
 
-bool isPermissionsPolicyAllowedByDocumentAndAllOwners(PermissionsPolicy::Type type, const Document& document, LogPermissionsPolicyFailure logFailure)
+static std::pair<PermissionsPolicy::Feature, StringView> splitFeatureName(StringView value)
```

Comment:
Add https://w3c.github.io/webappsec-permissions-policy/#serialized-policy-directive link?
Maybe we should reuse the terminology as well.